Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift

```diff
@@ -79,13 +78,10 @@ final class CacheHeaderHandleTests: XCTestCase {
         XCTAssertNotEqual(headerHandle1.offsetInFileOfOldestMessage, defaultOffsetInFileOfOldestMessage)
 
         let headerHandle2 = try createHeaderHandle(maximumBytes: 10000)
-        try headerHandle2.synchronizeHeaderData()
-
-        XCTAssertEqual(headerHandle2.offsetInFileAtEndOfNewestMessage, defaultOffsetInFileAtEndOfNewestMessage)
```

Comment:
It could be nice to have some more guarantees around what should happen here.
Beyond not throwing I think it would be nice to know if the header gets rewritten or not.

Reply:
yeah I can ensure the file doesn't change. good call